src/app/libp2p_helper/src/codanet.go

```diff
@@ -209,57 +211,6 @@ func (cm *CodaConnectionManager) Connected(net network.Network, c network.Conn)
 	logger.Debugf("%s connected to %s", c.LocalPeer(), c.RemotePeer())
 	cm.OnConnect(net, c)
 	cm.p2pManager.Notifee().Connected(net, c)
-
-	info := cm.GetInfo()
-	if len(net.Peers()) <= info.HighWater {
-		return
-	}
-
-	cm.TrimOpenConns(context.Background())
-
-	if !cm.minaPeerExchange {
-		return
-	}
-
-	logger.Debugf("node=%s disconnecting from peer=%s; max peers=%d peercount=%d", c.LocalPeer(), c.RemotePeer(), info.HighWater, len(net.Peers()))
-
-	defer func() {
-		go func() {
-			// small delay to allow for remote peer to read from stream
-			time.Sleep(time.Millisecond * 400)
-			_ = c.Close()
-		}()
-	}()
-
-	// select random subset of our peers to send over, then disconnect
-	if cm.getRandomPeers == nil {
-		logger.Error("getRandomPeers function not set")
-		return
-	}
-
-	peers := cm.getRandomPeers(info.LowWater, c.RemotePeer())
-	bz, err := json.Marshal(peers)
-	if err != nil {
-		logger.Error("failed to marshal peers", err)
-		return
-	}
-
-	stream, err := cm.host.NewStream(cm.ctx, c.RemotePeer(), pxProtocolID)
```

Comment:
If we are getting rid of the custom peer exchange protocol, we should delete the other code that is defining and configuring the protocol. But I'm curious if we should consider keeping this in some capacity. Perhaps do the peer exchange when we trim a peer (which was how it should have been implemented originally anyway).

We can still try and take this out for now if we wish to move away from it, but our tests so far have been in environments where peers are discoverable over mDNS, and were thus insufficient to determine if removing seed peer exchange is safe.

Reply:
Nathan and I had discussion around the matter and decided that now that grace period is 15 seconds, there is no need to have a special random peers mechanism and one can safely rely on DHT-provided discovery.

Remaining bits of peer exchange protocol will be removed in further commits within this PR or in a later PR if we decide to merge this PR early.